compiler_gym/leaderboard/llvm_instcount.py

```diff
@@ -84,61 +87,69 @@
 # interacts with that environment with the goal of maximising cumulative reward.
 Policy = Callable[[LlvmEnv], None]
 
+def eval_policy(d):
+    benchmark, policy = d
+    try:
+        env = gym.make("llvm-ic-v0")
+        env.logger.setLevel(logging.DEBUG)
 
-class _EvalPolicyWorker(Thread):
+        # print(f"Start one job {benchmark}!", flush=True)
+        env.reset(benchmark=benchmark)
+        with Timer() as timer:
+            policy(env)
+
+        # Sanity check that the policy didn't change the expected
+        # experimental setup.
+        assert env.in_episode, "Environment is no longer in an episode"
+        assert env.benchmark and (
+            env.benchmark == benchmark
+        ), "Policy changed environment benchmark"
+        assert env.reward_space, "Policy unset environment reward space"
+        assert (
+            env.reward_space.id == "IrInstructionCountOz"
+        ), "Policy changed environment reward space"
+
+        # print("Finish one job!", flush=True)
+        # Override walltime in the generated state.
+        state = env.state.copy()
+        state.walltime = timer.time
+        env.close()
+        return benchmark, state
+    except:
+        # If the code fails, then return None and we restart the same job again.
+        return benchmark, None
```

Comment:
I would remove this bare `except` and let the exception propagate. Is there a particular type of error you're interested in?